src/utils/env.ts

```diff
@@ -65,3 +87,24 @@ export const lookupPcEnv = (pcEnvTld: string, lenient = false, envTlds: string[]
 
     return null;
 };
+
+/**
+ * Attempts to locate a PC environment corresponding to the provided url
+ * @param url A string representing the Genesys Cloud environment top-level domain to search for
+ * @returns A Genesys Cloud environment object if found; null otherwise.
+ */
+export const lookupGcEnv = (url: string, targetEnv: string, parseEnvironment: EnvironmentParser = parse): PcEnv|null => {
+    try {
+        const hostLocation = new URL(url);
+        if (['localhost', '127.0.0.1'].includes(hostLocation.hostname)) {
```

Comment:
hmm ... wonder if we should validate the targetEnv in this case.  Probably makes sense to given that apps will likely use that env to set the client api for public api.

Maybe we should just add a validate/assert targetEnv util method that checks if it's valid (against sd-web envs and our env-specified envs) before we even bother asserting the url?